### ISSUE TITLE
`aws_cognito_user_pool` not refreshing schema from aws

### ISSUE THREAD

**Issue by nick-woodward** [terraform-providers/terraform-provider-aws#3405]
### Terraform Version

v0.11.3

### Affected Resource(s)

- aws_cognito_user_pool

### Terraform Configuration Files

```hcl
resource "aws_cognito_user_pool" "pool" {
  name = "dummy-user-pool"
  
  schema {
    name = "email"
    attribute_data_type = "String"
    required = true
  }

  auto_verified_attributes = [
    "email"
  ]
}
```

### Debug Output

https://gist.github.com/nick-woodward/90bd0c82cb1244e91c1acbafbd09cb54

### Expected Behavior

Terraform should have detected that the schema didn't change.

### Actual Behavior

Terraform marks the `aws_cognito_user_pool ` for deletion.

### Steps to Reproduce

1. Create a user pool manually that matches the above configuration.
2. `terraform import aws_cognito_user_pool.pool <id>`
3. `terraform apply`

**Comment by loivis** [comment]
What is the output from `terraform plan`? Does this match exactly what you created manually?

```
{
    "UserPool": {
        "AdminCreateUserConfig": {
            "AllowAdminCreateUserOnly": false,
            "InviteMessageTemplate": {
                "EmailMessage": "Your username is {username} and temporary password is {####}. ",
                "EmailSubject": "Your temporary password",
                "SMSMessage": "Your username is {username} and temporary password is {####}. "
            },
            "UnusedAccountValidityDays": 7
        },
        "AutoVerifiedAttributes": [
            "email"
        ],
        "CreationDate": 1.518725886449E9,
        "EmailConfiguration": {},
        "EmailVerificationMessage": "Your verification code is {####}. ",
        "EmailVerificationSubject": "Your verification code",
        "EstimatedNumberOfUsers": 0,
        "Id": "us-east-1_pGnrGfV9Q",
        "LambdaConfig": {},
        "LastModifiedDate": 1.518725887232E9,
        "MfaConfiguration": "OFF",
        "Name": "dummy-user-pool",
        "Policies": {
            "PasswordPolicy": {
                "MinimumLength": 8,
                "RequireLowercase": true,
                "RequireNumbers": true,
                "RequireSymbols": true,
                "RequireUppercase": true
            }
        },
        "SchemaAttributes": [
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": false,
                "Name": "sub",
                "Required": true,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "1"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "name",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "given_name",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "family_name",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "middle_name",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "nickname",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "preferred_username",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "profile",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "picture",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "website",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "email",
                "Required": true,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "Boolean",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "email_verified",
                "Required": false
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "gender",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "birthdate",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "10",
                    "MinLength": "10"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "zoneinfo",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "locale",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "phone_number",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "Boolean",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "phone_number_verified",
                "Required": false
            },
            {
                "AttributeDataType": "String",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "address",
                "Required": false,
                "StringAttributeConstraints": {
                    "MaxLength": "2048",
                    "MinLength": "0"
                }
            },
            {
                "AttributeDataType": "Number",
                "DeveloperOnlyAttribute": false,
                "Mutable": true,
                "Name": "updated_at",
                "NumberAttributeConstraints": {
                    "MinValue": "0"
                },
                "Required": false
            }
        ],
        "SmsAuthenticationMessage": "Your authentication code is {####}. ",
        "SmsVerificationMessage": "Your verification code is {####}. ",
        "UserPoolAddOns": {},
        "UserPoolTags": {},
        "VerificationMessageTemplate": {
            "DefaultEmailOption": "CONFIRM_WITH_CODE",
            "EmailMessage": "Your verification code is {####}. ",
            "EmailSubject": "Your verification code",
            "SmsMessage": "Your verification code is {####}. "
        }
    }
}
```

**Comment by nick-woodward** [comment]
@loivis Here's the output from `terraform plan`.

```sh
$ terraform plan

Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

aws_cognito_user_pool.pool: Refreshing state... (ID: us-east-1_qtrBCU56Z)

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

-/+ aws_cognito_user_pool.pool (new resource required)
      id:                                               "us-east-1_qtrBCU56Z" => <computed> (forces new resource)
      admin_create_user_config.#:                       "1" => <computed>
      arn:                                              "arn:aws:cognito-idp:us-east-1:378350985039:userpool/us-east-1_qtrBCU56Z" => <computed>
      auto_verified_attributes.#:                       "1" => "1"
      auto_verified_attributes.881205744:               "email" => "email"
      creation_date:                                    "2018-02-16T15:03:01Z" => <computed>
      email_verification_message:                       "" => <computed>
      email_verification_subject:                       "" => <computed>
      lambda_config.#:                                  "0" => <computed>
      last_modified_date:                               "2018-02-16T15:03:01Z" => <computed>
      mfa_configuration:                                "OFF" => "OFF"
      name:                                             "dummy-user-pool" => "dummy-user-pool"
      password_policy.#:                                "1" => <computed>
      schema.#:                                         "" => "1" (forces new resource)
      schema.3593246171.attribute_data_type:            "" => "String"
      schema.3593246171.developer_only_attribute:       "" => ""
      schema.3593246171.mutable:                        "" => ""
      schema.3593246171.name:                           "" => "email"
      schema.3593246171.number_attribute_constraints.#: "" => "0"
      schema.3593246171.required:                       "" => "true"
      schema.3593246171.string_attribute_constraints.#: "" => "0"
      verification_message_template.#:                  "1" => <computed>


Plan: 1 to add, 0 to change, 1 to destroy.

------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.
```

**Comment by loivis** [comment]
Took a quick look at code but didn't find any handling of `schema` while reading  aws resource. I think it is something missing. Can you confirm `schema` doesn't exist in terraform state file after import?

**Comment by nick-woodward** [comment]
Here's the contents of the `terraform.tfstate` file after importing the user pool.

```json
{
    "version": 3,
    "terraform_version": "0.11.3",
    "serial": 1,
    "lineage": "bd5ba2d5-3acf-49eb-9cf1-212eb157ca47",
    "modules": [
        {
            "path": [
                "root"
            ],
            "outputs": {},
            "resources": {
                "aws_cognito_user_pool.pool": {
                    "type": "aws_cognito_user_pool",
                    "depends_on": [],
                    "primary": {
                        "id": "us-east-1_BiuURK40P",
                        "attributes": {
                            "admin_create_user_config.#": "1",
                            "admin_create_user_config.0.allow_admin_create_user_only": "false",
                            "admin_create_user_config.0.invite_message_template.#": "0",
                            "admin_create_user_config.0.unused_account_validity_days": "7",
                            "arn": "arn:aws:cognito-idp:us-east-1:378350985039:userpool/us-east-1_BiuURK40P",
                            "auto_verified_attributes.#": "1",
                            "auto_verified_attributes.881205744": "email",
                            "creation_date": "2018-02-16T17:32:08Z",
                            "device_configuration.#": "0",
                            "email_configuration.#": "0",
                            "id": "us-east-1_BiuURK40P",
                            "lambda_config.#": "0",
                            "last_modified_date": "2018-02-16T17:32:08Z",
                            "mfa_configuration": "OFF",
                            "name": "dummy-user-pool",
                            "password_policy.#": "1",
                            "password_policy.0.minimum_length": "8",
                            "password_policy.0.require_lowercase": "true",
                            "password_policy.0.require_numbers": "true",
                            "password_policy.0.require_symbols": "true",
                            "password_policy.0.require_uppercase": "true",
                            "sms_configuration.#": "0",
                            "tags.%": "0",
                            "verification_message_template.#": "1",
                            "verification_message_template.0.default_email_option": "CONFIRM_WITH_CODE",
                            "verification_message_template.0.email_message": "",
                            "verification_message_template.0.email_message_by_link": "",
                            "verification_message_template.0.email_subject": "",
                            "verification_message_template.0.email_subject_by_link": "",
                            "verification_message_template.0.sms_message": ""
                        },
                        "meta": {},
                        "tainted": false
                    },
                    "deposed": [],
                    "provider": "provider.aws"
                }
            },
            "depends_on": []
        }
    ]
}
```

**Comment by bflad** [comment]
Bugfix PR submitted: #3789

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with v1.13.0 of the AWS provider, likely at the end of next week.

**Comment by bflad** [comment]
Ahead of schedule, this has been released in [version 1.13.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1130-march-28-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by Puneeth-n** [comment]
@bflad This change is destroying and recreating the pool. Any Idea what is happening with me here?

```
-/+ aws_cognito_user_pool.user_pool (new resource required)
      id:                                                                 "eu-west-1_lq5VDwN7G" => <computed> (forces new resource)
      admin_create_user_config.#:                                         "1" => "1"
      admin_create_user_config.0.allow_admin_create_user_only:            "false" => "false"
      admin_create_user_config.0.invite_message_template.#:               "0" => "1"
      admin_create_user_config.0.invite_message_template.0.email_message: "" => "Your username is {username} and temporary password is {####}."
      admin_create_user_config.0.invite_message_template.0.email_subject: "" => "Your temporary password"
      admin_create_user_config.0.invite_message_template.0.sms_message:   "" => "Your username is {username} and temporary password is {####}."
      admin_create_user_config.0.unused_account_validity_days:            "7" => "7"
      arn:                                                                "arn:aws:cognito-idp:eu-west-1:0123456789:userpool/eu-west-1_lq5VDwN7G" => <computed>
      auto_verified_attributes.#:                                         "0" => "1"
      auto_verified_attributes.881205744:                                 "" => "email"
      creation_date:                                                      "2018-03-08T09:26:39Z" => <computed>
      device_configuration.#:                                             "0" => "1"
      device_configuration.0.challenge_required_on_new_device:            "" => "false"
      device_configuration.0.device_only_remembered_on_user_prompt:       "" => "true"
      email_configuration.#:                                              "1" => "1"
      email_configuration.0.source_arn:                                   "arn:aws:ses:eu-west-1:0123456789:identity/no-reply@comtravo.com" => "arn:aws:ses:eu-west-1:0123456789:identity/no-reply@comtravo.com"
      email_verification_message:                                         "" => <computed>
      email_verification_subject:                                         "" => <computed>
      lambda_config.#:                                                    "1" => "1"
      lambda_config.0.custom_message:                                     "arn:aws:lambda:eu-west-1:0123456789:function:lambda-cognito-trigger-verification_email-beta" => "${module.lambda-cognito-trigger.lambda_arn}"
      last_modified_date:                                                 "2018-04-05T09:30:09Z" => <computed>
      mfa_configuration:                                                  "OFF" => "OFF"
      name:                                                               "my-comtravo-beta" => "my-comtravo-beta"
      password_policy.#:                                                  "1" => "1"
      password_policy.0.minimum_length:                                   "8" => "8"
      password_policy.0.require_lowercase:                                "true" => "true"
      password_policy.0.require_numbers:                                  "true" => "true"
      password_policy.0.require_symbols:                                  "false" => "false"
      password_policy.0.require_uppercase:                                "true" => "true"
      schema.#:                                                           "4" => "4"
      schema.1075931092.attribute_data_type:                              "Number" => "" (forces new resource)
      schema.1075931092.developer_only_attribute:                         "false" => "false"
      schema.1075931092.mutable:                                          "true" => "false" (forces new resource)
      schema.1075931092.name:                                             "custom:can_book_for_others" => "" (forces new resource)
      schema.1075931092.number_attribute_constraints.#:                   "1" => "0" (forces new resource)
      schema.1075931092.number_attribute_constraints.0.max_value:         "1" => "" (forces new resource)
      schema.1075931092.number_attribute_constraints.0.min_value:         "0" => "" (forces new resource)
      schema.1075931092.required:                                         "false" => "false"
      schema.1075931092.string_attribute_constraints.#:                   "0" => "0"
      schema.1117227757.attribute_data_type:                              "" => "String" (forces new resource)
      schema.1117227757.developer_only_attribute:                         "" => "false" (forces new resource)
      schema.1117227757.mutable:                                          "" => "true" (forces new resource)
      schema.1117227757.name:                                             "" => "traveler_id" (forces new resource)
      schema.1117227757.number_attribute_constraints.#:                   "" => "0"
      schema.1117227757.required:                                         "" => "false" (forces new resource)
      schema.1117227757.string_attribute_constraints.#:                   "" => "1" (forces new resource)
      schema.1117227757.string_attribute_constraints.0.max_length:        "" => "256" (forces new resource)
      schema.1117227757.string_attribute_constraints.0.min_length:        "" => "1" (forces new resource)
      schema.1374720567.attribute_data_type:                              "" => "Number" (forces new resource)
      schema.1374720567.developer_only_attribute:                         "" => "false" (forces new resource)
      schema.1374720567.mutable:                                          "" => "true" (forces new resource)
      schema.1374720567.name:                                             "" => "can_book_for_others" (forces new resource)
      schema.1374720567.number_attribute_constraints.#:                   "" => "1" (forces new resource)
      schema.1374720567.number_attribute_constraints.0.max_value:         "" => "1" (forces new resource)
      schema.1374720567.number_attribute_constraints.0.min_value:         "" => "0" (forces new resource)
      schema.1374720567.required:                                         "" => "false" (forces new resource)
      schema.1374720567.string_attribute_constraints.#:                   "" => "0"
      schema.1759222595.attribute_data_type:                              "Number" => "" (forces new resource)
      schema.1759222595.developer_only_attribute:                         "false" => "false"
      schema.1759222595.mutable:                                          "true" => "false" (forces new resource)
      schema.1759222595.name:                                             "custom:admin" => "" (forces new resource)
      schema.1759222595.number_attribute_constraints.#:                   "1" => "0" (forces new resource)
      schema.1759222595.number_attribute_constraints.0.max_value:         "1" => "" (forces new resource)
      schema.1759222595.number_attribute_constraints.0.min_value:         "0" => "" (forces new resource)
      schema.1759222595.required:                                         "false" => "false"
      schema.1759222595.string_attribute_constraints.#:                   "0" => "0"
      schema.2143646793.attribute_data_type:                              "String" => "" (forces new resource)
      schema.2143646793.developer_only_attribute:                         "false" => "false"
      schema.2143646793.mutable:                                          "true" => "false" (forces new resource)
      schema.2143646793.name:                                             "custom:company_id" => "" (forces new resource)
      schema.2143646793.number_attribute_constraints.#:                   "0" => "0"
      schema.2143646793.required:                                         "false" => "false"
      schema.2143646793.string_attribute_constraints.#:                   "1" => "0" (forces new resource)
      schema.2143646793.string_attribute_constraints.0.max_length:        "256" => "" (forces new resource)
      schema.2143646793.string_attribute_constraints.0.min_length:        "1" => "" (forces new resource)
      schema.2902373836.attribute_data_type:                              "" => "String" (forces new resource)
      schema.2902373836.developer_only_attribute:                         "" => "false" (forces new resource)
      schema.2902373836.mutable:                                          "" => "true" (forces new resource)
      schema.2902373836.name:                                             "" => "company_id" (forces new resource)
      schema.2902373836.number_attribute_constraints.#:                   "" => "0"
      schema.2902373836.required:                                         "" => "false" (forces new resource)
      schema.2902373836.string_attribute_constraints.#:                   "" => "1" (forces new resource)
      schema.2902373836.string_attribute_constraints.0.max_length:        "" => "256" (forces new resource)
      schema.2902373836.string_attribute_constraints.0.min_length:        "" => "1" (forces new resource)
      schema.3186168095.attribute_data_type:                              "" => "Number" (forces new resource)
      schema.3186168095.developer_only_attribute:                         "" => "false" (forces new resource)
      schema.3186168095.mutable:                                          "" => "true" (forces new resource)
      schema.3186168095.name:                                             "" => "admin" (forces new resource)
      schema.3186168095.number_attribute_constraints.#:                   "" => "1" (forces new resource)
      schema.3186168095.number_attribute_constraints.0.max_value:         "" => "1" (forces new resource)
      schema.3186168095.number_attribute_constraints.0.min_value:         "" => "0" (forces new resource)
      schema.3186168095.required:                                         "" => "false" (forces new resource)
      schema.3186168095.string_attribute_constraints.#:                   "" => "0"
      schema.3751200387.attribute_data_type:                              "String" => "" (forces new resource)
      schema.3751200387.developer_only_attribute:                         "false" => "false"
      schema.3751200387.mutable:                                          "true" => "false" (forces new resource)
      schema.3751200387.name:                                             "custom:traveler_id" => "" (forces new resource)
      schema.3751200387.number_attribute_constraints.#:                   "0" => "0"
      schema.3751200387.required:                                         "false" => "false"
      schema.3751200387.string_attribute_constraints.#:                   "1" => "0" (forces new resource)
      schema.3751200387.string_attribute_constraints.0.max_length:        "256" => "" (forces new resource)
      schema.3751200387.string_attribute_constraints.0.min_length:        "1" => "" (forces new resource)
      sms_verification_message:                                           "Your verification code is {####}." => "Your verification code is {####}."
      tags.%:                                                             "2" => "2"
      tags.Name:                                                          "my-comtravo-beta" => "my-comtravo-beta"
      tags.environment:                                                   "beta" => "beta"
      username_attributes.#:                                              "1" => "1"
      username_attributes.0:                                              "email" => "email"
      verification_message_template.#:                                    "1" => <computed>

```
